### PR TITLE
ddl: speed up the operation of "drop database"

### LINKED ISSUE #16983: Drop a database hang more then 1 hour
## Bug Report

When drop a database that have more then 1.6M table. Then it takes 1h50m to drop databases 

### 1. What did you do?
`drop database multi_pk_2;`

### 2. What did you expect to see?
It takes a few minutes to drop a database

### 3. What did you see instead?
It takes 1h50m to drop a database

### 4. What version of TiDB are you using? (`tidb-server -V` or run `select tidb_version();` on TiDB)
```
Release Version: v4.0.0-beta.2-352-g5ede18f10
Git Commit Hash: 5ede18f10eedfe2e3690d7728dec3ffa4b0af2d5
Git Branch: master
UTC Build Time: 2020-04-24 03:45:17
GoVersion: go1.13
Race Enabled: false
TiKV Min Version: v3.0.0-60965b006877ca7234adaced7890d7b029ed1306
Check Table Before Drop: false
```

### PR DESCRIPTION
**Pull request by zimulala**
### What problem does this PR solve?

Issue Number: close #16983 

Problem Summary:
Screen Shot 2020-05-07 at 12 42 14
Looking at the logs, this drop schema operation really took almost 1 hour. 
WeChatWorkScreenshot_b015a9fc-b256-47d1-82e6-8493c00d7890
The insert statements take more than 50 minutes. The other time consuming is due to the synchronization method of waiting for 2 * lease.
This PR we speed up the insert statements in "drop schema" operation.

### What is changed and how it works?

What's Changed:
In the above case, the database has 1587204 tables, so it needs to execute 1587204 insert statements. In this PR we do batch insertion.

### Check List 

Tests 

- Unit test (original tests, and set `BatchInsertDeleteRangeSize` to 2)
- Manual test (add detailed scripts or steps below)

Using TiUP to start a cluster for 3 tikv，1 tidb and 1 pd.

dbx* has 1 table, dby* has 100 tables, dbz* has 10000 tables.

Before this PR:
```
mysql> drop database dbx1;
Query OK, 0 rows affected (0.33 sec)
mysql> drop database dby1;
Query OK, 0 rows affected (0.69 sec)
mysql> drop database dbz1;
Query OK, 0 rows affected (55.89 sec)
```
After this PR:
```
mysql> drop database dbx0;
Query OK, 0 rows affected (0.27 sec)
mysql> drop database dby0;
Query OK, 0 rows affected (0.38 sec)
mysql> drop database dbz0;
Query OK, 0 rows affected (3.77 sec)
```

### Release note 

- Speed up the deletion of databases with a large number of tables

### CODE REVIEW

File: ddl/ddl_test.go [hunk 1]
```diff
@@ -110,6 +110,7 @@ func TestT(t *testing.T) {
 	logutil.InitLogger(logutil.NewLogConfig(logLevel, "", "", logutil.EmptyFileLogConfig, false))
 	autoid.SetStep(5000)
 	ReorgWaitTimeout = 30 * time.Millisecond
+	batchInsertDeleteRangeSize = 2
```
Comment (AilinKid) [inline ddl/ddl_test.go:113]: should add a benchmark test?

**Comment by bb7133** [comment]
Please address @djshow832 's comment, rest LGTM.

**Comment by zimulala** [comment]
PTAL @bb7133 @djshow832 PTAL

File: ddl/ddl_test.go (hunk 1 above)
Comment (zimulala) [reply ddl/ddl_test.go:113]: If the benchmarks use mockTiKV, I don't think it makes much sense.

Comment (zimulala) [reply ddl/ddl_test.go:113]: @AilinKid PTAL

**Comment by bb7133** [comment]
Failed cases:

```
FAIL: statistics_handler_test.go:86: testDumpStatsSuite.TestDumpStatsAPI

[2020-05-09T06:09:38.384Z] 

[2020-05-09T06:09:38.384Z] statistics_handler_test.go:87:

[2020-05-09T06:09:38.384Z]     ds.startServer(c)

[2020-05-09T06:09:38.384Z] statistics_handler_test.go:64:

[2020-05-09T06:09:38.384Z]     c.Assert(err, IsNil)

[2020-05-09T06:09:38.384Z] ... value *errors.withStack = listen tcp 0.0.0.0:10092: bind: address already in use ("listen tcp 0.0.0.0:10092: bind: address already in use")
```
https://internal.pingcap.net/idc-jenkins/blue/organizations/jenkins/tidb_ghpr_unit_test/detail/tidb_ghpr_unit_test/35674/

**Comment by zimulala** [comment]
@bb7133 This test case is unstable, we have an issue: #15600

**Pull request merged**